### ISSUE TITLE
1.0.12 fires stop BEFORE drop. 1.0.6 fires stop AFTER drop

### ISSUE THREAD

**Issue by poshest** [codef0rmer/angular-dragdrop#245]
Any idea why this changed?  I upgraded to 1.0.12 and it broke my code - clean up I was doing in `stop` was removing stuff I needed in `drop`

**Comment by codef0rmer** [comment]
@poshest A demo would be helpful to understand the problem

**Comment by codef0rmer** [comment]
Duplicate of #241.

**Comment by poshest** [comment]
This is not a duplicate of #241.  It has nothing to do with different versions of Angular. Re the "demo" do you have a jsfiddle already set up with the basics, so that I can copy it?

**Comment by codef0rmer** [comment]
Goto any example and fork the plunkr at http://codef0rmer.github.io/angular-dragdrop/#/

**Comment by codef0rmer** [comment]
@poshest: BTW I presumed it to be duplicate of #241 because I though you might have run the digest loop in the drop callback which eventually resolve the promise of beforeDrop. 

I would be interested to understand the problem by a demo?

**Comment by poshest** [comment]
This was as far as I got with a [Plunkr based on 1.0.6](http://plnkr.co/edit/W9p4EsqFS2vVssIOAqHT?p=preview), but I've linked it only for reference.  I was unable to reproduce the drop before stop in the Plunkr that I'm seeing in my code.  

In any case, it's clear that stop before drop is the standard, and I should just upgrade to 1.0.12 and handle the error some other way.  

As a side note, I don't understand why anyone would *want* stop before drop - I use drop to process the dropped object, and stop to clean up the dropped object, which I thought would be the most common use case. But there are obviously good reasons to do stop then drop.
